### PR TITLE
Bug/integration tests

### PR DESCRIPTION
**Pull request by kaushik94**
Integration tests

## Description

Integrations tests that ingest a config.yml and do a Simulation test. Very useful for reviewing PRs and make sure no breaking changes.

## Motivation and Context

These are very advanced tests and necessary to check the accuracy of TARDIS performance from time to time. It has been discussed several times and I have been working on it for over a year now.
The breaking changes and issues it addresses are:
#1307 
#997 

## How Has This Been Tested?

I tested this by running the integration_tests for w7 paper. The tests fail as expected because the v_boundary is not set correctly. Plus there is insufficient data and very few packets. Therefore the accuracy is very low. But I hope we can get this working so that integration tests can at-least be run a few days on a server.
This PR is not intended for merging directly as it sets `if ( (nu_diff = comov_nu - nu_line) > 0)`. This has already been discussed in the meeting as far as I remember but I do not know if setting > 0 helps.

## Screenshots (if appropriate):
Screenshot 2020-10-09 at 6 47 50 PM

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have assigned/requested two reviewers for this pull request.

### CODE REVIEW

File: tardis/tests/integration_tests/conftest.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import pytest
 
 from tardis import __githash__ as tardis_githash
-from tardis.tests.integration_tests.report import DokuReport
+#from tardis.tests.integration_tests.report import DokuReport
 from tardis.tests.integration_tests.plot_helpers import (
```
Comment (kaushik94) [inline tardis/tests/integration_tests/conftest.py:9]: @epassaro removing dokureport dependency will solve the issue of `self.contained` error. But now the data seems to be wrong as it gives error related to v_boundary being large.

**Comment by epassaro** [comment]
I'm reviewing this @kaushik94

**Comment by epassaro** [comment]
I'm not being able to reproduce that error. Are you trying to run just the `w7` integration test, right? Can you paste the command you use to run that test? @kaushik94

**Comment by kaushik94** [comment]
@epassaro go into the `integration_tests` directory and run the following:
`pytest --integration-tests=integration.yml -m integration --less-packets`
Yes for now it only works for w7

File: tardis/tests/integration_tests/w7/config.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-atom_data: kurucz_cd23_chianti_H_He.h5
+atom_data: atom_data/kurucz_cd23_chianti_H_He.h5
```
Comment (kaushik94) [inline tardis/tests/integration_tests/w7/config.yml:1]: @epassaro make sure this is the correct directory.

File: tardis/tests/integration_tests/integration.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+atom_data_path: "~/Desktop/tardis/tardis-refdata/"
```
Comment (kaushik94) [inline tardis/tests/integration_tests/integration.yml:1]: @epassaro please change this to point to your refdata. As well as some other lines in this file.

**Comment by epassaro** [comment]
Yes, I pointed to the right atom data file, but I get some validation schema error which is different from yours :/ 

I'll think about it! :smiley:

**Comment by kaushik94** [comment]
@epassaro try running:
`pytest --integration-tests=integration.yml -m integration --less-packets --generate-reference`
I'm not sure if this will solve it.

**Comment by kaushik94** [comment]
@epassaro oh yes now I remember. You need to move `w7.h5` file into `integration_tests` folder. Currently it is at `integration_tests/w7` folder.

**Comment by epassaro** [comment]
> @epassaro oh yes now I remember. You need to move `w7.h5` file into `integration_tests` folder. Currently it is at `integration_tests/w7` folder.

I couldn't generate the`w7.h5` file with `--generate-reference`. Or I am doing something wrong?

**Comment by kaushik94** [comment]
@epassaro yes I understood, no not your mistake. It is actually in the refdata folder. You need to copy that into the root of integration_tests folder.

**Comment by andrewfullard** [comment]
Is this PR relevant now that we have moved to numba? I think it needs to be at least partly rewritten.

**Comment by kaushik94** [comment]
@andrewfullard I'm not sure this is relevant once the C code is replaced with numba. It would be great if I have an overview of the main functions in numba to test. The test interface might be similar but the I/O would change. Thanks.

**Comment by andrewfullard** [comment]
> 
> 
> @andrewfullard I'm not sure this is relevant once the C code is replaced with numba. It would be great if I have an overview of the main functions in numba to test. The test interface might be similar but the I/O would change. Thanks.

@kaushik94 there is already a test framework for Numba in tardis/montecarlo/numba_montecarlo/tests. There are quite a few empty tests there that could probably be filled from your tests here with some modification.

**Comment by kaushik94** [comment]
@andrewfullard awesome. Thanks for mentioning. I will look into it.

**Comment by kaushik94** [comment]
@andrewfullard I will open a new one for the Numba code testing. Closing this now.

**Pull request closed without merging**